src/app/cds-fetcher.service.ts

```diff
@@ -59,17 +62,17 @@ function parseTimeseriesData(tsData: CDSLocationTimeSeriesData): Foo {
   providedIn: 'root',
 })
 export class CdsFetcherService {
-  data: Promise<{ [key: string]: Foo }>;
+  data: Observable<{ [key: string]: Foo }>;
 
   constructor() {
-    const CORONA_URL = `https://coronadatascraper.com/timeseries-byLocation.json`;
-    let timeInit;
-    this.data = fetch(CORONA_URL)
-      .then(response => {
-        timeInit = performance.now();
-
-        return response.json();
-      })
-      .then(mapValues(parseTimeseriesData));
+    this.data = new Observable((observer) => {
+      fetch(CORONA_URL)
+        // get response data to json
+        .then((res) => res.json())
+        // convert json to correctly formatted timeseries
+        .then(mapValues(parseTimeseriesData))
+        // emit the processed data.
+        .then((data) => observer.next(data));
+    });
```

Comment:
Optional change: Instead of manually creating an Observable, you can use rxjs's `from()`!

(Also, if you've never seen this, it's quite handy: https://rxjs.dev/operator-decision-tree)

Reply:
this is awesome